src/encode.cpp

```diff
@@ -277,4 +278,48 @@ operator>>(MessageBuffer& buffer, PublishIntentEnd& msg)
   return buffer;
 }
 
+messages::MessageBuffer&
+operator<<(messages::MessageBuffer& msg, const quicr::Name& val)
+{
+  constexpr uint8_t size = sizeof(quicr::Name::uint_type) * 2;
+  for (size_t i = 0; i < size; ++i)
+    msg << val[size - 1 - i];
```

Comment:
Just ran some benchmarks on this.

Before this change:
```benchmark
--------------------------------------------------------------
Benchmark                    Time             CPU   Iterations
--------------------------------------------------------------
BM_NameEncodeDecode       2089 ns         2089 ns       322659
```

With this change:
```benchmark
--------------------------------------------------------------
Benchmark                    Time             CPU   Iterations
--------------------------------------------------------------
BM_NameEncodeDecode        975 ns          975 ns       623564
```

And for completeness, testing with just writing `val._low` and `val._hi` straight to the buffer:
```benchmark
--------------------------------------------------------------
Benchmark                    Time             CPU   Iterations
--------------------------------------------------------------
BM_NameEncodeDecode        636 ns          636 ns       823045

Reply:
Here's the benchmark code:
```cpp
    quicr::Name encode("0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF");
    quicr::Name decode;
    quicr::messages::MessageBuffer buffer;
    for (auto _ : state)
    {
        buffer << encode;
        buffer >> decode;
    }

Reply:
This is with google benchmark